jbmc/unit/java_bytecode/goto_program_generics/generic_bases_test.cpp

```diff
@@ -469,17 +471,18 @@ SCENARIO(
       // We trace the creation of the object that is being supplied as
       // the input to the method under test. There must be one non-null
       // assignment only, and usually looks like this:
-      //   this = &tmp_object_factory$1;
+      //   this = malloc_site;
       const irep_idt &this_tmp_name =
         require_goto_statements::require_entry_point_argument_assignment(
           ID_this, entry_point_code);
 
       THEN("Object 'this' created has unspecialized inherited field")
       {
-        //   tmp_object_factory$1.@UnsupportedWrapper1.field =
-        // &tmp_object_factory$2;
-        // struct java.lang.Object { __CPROVER_string @class_identifier; }
-        // tmp_object_factory$2;
+        // malloc_site->@UnsupportedWrapper1.field = malloc_site$0;
```

Comment:
The comment is giving what code we expect to find. That has changed because `this_tmp_name` comes `require_entry_point_argument_assignment()`, and entry point arguments are now assigned differently.

Reply:
Yes. I should make the comment clearer about what's going on. Also, I think I'm trying to explain too much in the wrong place.